### ISSUE TITLE
Yasnipet is broken in dart mode

### ISSUE THREAD

**Issue by itome** [emacs-lsp/lsp-mode#1337]
yasnippet with lsp-mode works fine with dart-mode just after lsp-mode started, but sometimes (I don't know the reason) error occurs in completion/resolve and after that, snippet directly shown in buffer like below.

<img width="331" alt="Screen Shot 2020-01-20 at 18 46 37" src="https://user-images.githubusercontent.com/24409457/72716252-402c8400-3bb5-11ea-9962-9cc08cc6904b.png">

First error message is

```
Error in post-command-hook (yas--post-command-handler): (wrong-type-argument markerp 2268)
```

And after the error, ↓ this error occur every time snippet expand.

```
company-call-backend-raw: Company: backend company-lsp error "[yas] ‘yas-expand-snippet’ needs properly setup ‘yas-minor-mode’" with args (post-completion child: )
company-call-backend-raw: Company: backend company-lsp error "[yas] ‘yas-expand-snippet’ needs properly setup ‘yas-minor-mode’" with args (post-completion borderRadius: )
company-call-backend-raw: Company: backend company-lsp error "[yas] ‘yas-expand-snippet’ needs properly setup ‘yas-minor-mode’" with args (post-completion floatingActionButton: )
```

**Comment by yyoncho** [comment]
We are now focusing on making capf interface but probably it will have the same issue. Next time when it happen, can you turn on the client server logging and then redo the same completion and then attach the logs?

**Comment by itome** [comment]
@yyoncho This is log from `lsp-log-io` after snippet broken. This helps you?

https://gist.github.com/itome/071082ab062b7d851263d67eb3500f2b

**Comment by itome** [comment]
@yyoncho Has any progress been made? This bug seems to reproducible in current release.

**Comment by yyoncho** [comment]
@itome sorry, I haven't had the time to check it: does this reproduces with company-capf? Also, do you have exact steps + demo project?

**Comment by itome** [comment]
This behavior occurs by chance. So I can't specify the project.
I think this issue is same as -> https://github.com/syl20bnr/spacemacs/issues/13358

**Comment by itome** [comment]
This issue seems to be fixed after this https://github.com/syl20bnr/spacemacs/pull/13455.

**Comment by itome** [comment]
Sorry, this issue still occurs.

**Comment by yyoncho** [comment]
@ericdallo willing to take a look?

**Comment by ericdallo** [comment]
I could not reproduce this using `doom-emacs` with `company-capf` + `company-lsp`, and personally I never saw this behavior before in any `lsp`.
Maybe this should be a bug of how `spacemacs` configure `company`.

I suggest you check how `doom-emacs` [configure company](https://github.com/hlissner/doom-emacs/blob/develop/modules/completion/company/config.el) and try to figure out if there's some different config that not may cause this.

**Comment by billop** [comment]
I'm getting the same issue with rust-analyzer, whether I use company-lsp or company-capf.

**Comment by yyoncho** [comment]
@billop there is FAQ entry for your issue - the RA does not support disabling snippets.

**Comment by billop** [comment]
> @billop there is FAQ entry for your issue - the RA does not support disabling snippets.

I don't think it mentions enabling yas-minor-mode however. Not sure if that's too obvious for most people but it was not obvious to me.

**Comment by yyoncho** [comment]
Is this still happening?
